chore: fix publish ci, update .gitignore

josxha opened the pull request:
- use flutter in publish CI (seems to work fine)
- fix syntax error on called workflow action (publish-single.yml)
- update .gitignore to get rid on the flutter pub publish warning
  - it seems that the .gitignore was previously copied over from the flutter/flutter repo
  - removes ~~and adds~~ one file to the repo

For more information, see the CI history here: https://github.com/josxha/flutter-maplibre-gl/actions/workflows/publish.yml

josxha committed:
fix ci

josxha committed:
fix ci

josxha committed:
fix ci

josxha committed:
fix ci

josxha committed:
use flutter to publish

josxha committed:
use own ci

josxha committed:
limit repo owner (currently me)

josxha committed:
use flutter pub get instead of dart pub get

josxha committed:
update .gitignore

josxha committed:
skip confirmation

josxha committed:
Update publish.yml

josxha committed:
set maplibre as repo owner

josxha committed:
Merge branch 'main' into test/publish

josxha committed:
fix branch name

kuhnroyal commented on .github/workflows/publish.yml:
```diff
@@ -10,16 +10,16 @@ permissions:
 jobs:
   publish_maplibre_gl_platform_interface:
-    uses: ./.github/workflows/publish_single.yml
+    uses: maplibre/flutter-maplibre-gl/.github/workflows/publish-single.yml@main
```
I think this change is not needed, only the dash in the filename was wrong :)

josxha replied:
I tested it but the ci failed

- the @main is needed, see https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181129759/workflow
- and the owner/repo seems to be required too, see https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181145594

kuhnroyal replied:
> [Invalid workflow file: .github/workflows/publish.yml#L13](https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181145594/workflow)
invalid value workflow reference: references to workflows must be prefixed with format 'owner/repository/' or './' for local workflows

josxha committed:
use action from same branch

https://github.com/maplibre/flutter-maplibre-gl/pull/424#discussion_r1608918715

josxha committed:
fix: invalid value workflow reference: no version specified

kuhnroyal commented on .github/workflows/publish.yml:
```diff
@@ -10,16 +10,16 @@ permissions:
 jobs:
   publish_maplibre_gl_platform_interface:
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
And no branch. See https://stackoverflow.com/a/76614274

josxha replied:
I'll try it

josxha replied:
fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

kuhnroyal commented on .github/workflows/publish.yml:
```diff
@@ -10,16 +10,16 @@ permissions:
 jobs:
   publish_maplibre_gl_platform_interface:
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

josxha replied:
fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

kuhnroyal commented on .github/workflows/publish.yml:
```diff
@@ -10,16 +10,16 @@ permissions:
   publish_maplibre_gl_web:
     needs: [publish_maplibre_gl_platform_interface]
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

josxha replied:
fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

kuhnroyal commented on .github/workflows/publish.yml:
```diff
@@ -10,16 +10,16 @@ permissions:
   publish_maplibre_gl:
     needs: [publish_maplibre_gl_platform_interface, publish_maplibre_gl_web]
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

josxha replied:
fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

kuhnroyal commented on example/macos/Flutter/GeneratedPluginRegistrant.swift:
not sure about this file, don't think it should be checked in tbh

josxha replied:
I can remove it and git ignore it. You are right doesn't seem that necessary to include it.

kuhnroyal replied:
lets do it and then ill merge.

josxha replied:
fixed in 3b2e2a2bc9b7ed7d54181b6c9e78cce57d1cf556

josxha committed:
use requested syntax

josxha committed:
remove and ignore GeneratedPluginRegistrant.swift

kuhnroyal commented on .github/workflows/publish-single.yml:
```diff
@@ -17,13 +17,14 @@ jobs:
+        run: flutter pub get
+        # --force skips the y/N confirmation
       - name: Publish to pub.dev
-        run: dart pub publish --force --skip-validation
```
We need to revert these changes, release is stuck here.

josxha replied:
- https://github.com/maplibre/flutter-maplibre-gl/pull/426
